helm/chart/maesh/templates/controller/hooks/cleanup-hook.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: maesh-cleanup
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Release.Name | quote}}
+    chart: {{ include "maesh.chartLabel" . | quote }}
+    release: {{ .Release.Name | quote}}
+    heritage: {{ .Release.Service | quote}}
+  annotations:
+    "helm.sh/hook": "pre-delete"
```

Comment:
Do you think there would be an issue with service accounts? Since the service account we use to run the cleanup is managed by helm.

Or do you think we should create its own service account? idk.

Reply:
I think that the running controller could create a shadow service just after the cleanup. Btw, it is a corner case.

Reply:
unless we bundle an RBAC with the hook? and then that way it can remove services after the controller has been deleted?